system/logaccess/Azure/LogAnalytics/CurlClient/AzureLogAnalyticsCurlClient.cpp

```diff
@@ -447,7 +474,14 @@ bool generateHPCCLogColumnstAllColumns(StringBuffer & kql, const char * colName)
         return false;
     }
 
-    kql.appendf("\n| extend hpcclogfields = extract_all(@\'^([0-9A-Fa-f]+)\\s+(OPR|USR|PRG|AUD|UNK)\\s+(DIS|ERR|WRN|INF|PRO|MET|UNK)\\s+(\\d{4}-\\d{2}-\\d{2}\\s\\d{2}:\\d{2}:\\d{2}\\.\\d+)\\s+(\\d+)\\s+(\\d+)\\s+(UNK|[A-Z]\\d{8}-\\d{6}(?:-\\d+)?)\\s+\\\"(.*)\\\"$', %s)[0]", colName);
+    StringBuffer sourceCol;
+    if (strcmp(colName, "LogMessage")==0)
+        sourceCol.set("tostring(LogMessage)");
```

Comment:
good question. Yes they do, L484 parses either toString(LogMessage), or the configured source column LogContent and makes the parsed content available in the hpcclogfields variable.
If the source column doesn't contain the actual HPCC log line, then L484-492 don't makes sense and would be blank.

Reply:
>If the source column doesn't contain the actual HPCC log line, then L484-492 don't makes sense and would be blank.

(now lines 502-511)

that's what I meant, if not applicable, then this regex is going to result in blank fields, but it (KQL) still going to go through the expense per line of running a regex and parsing the 'extend' command fruitlessly?

Reply:
Are you suggesting to bail out if the target column is not one of the two known valid columns? I wanted to be 100% configuration based, but there is a bit of hardcoded ALA knowledge. 

Beyond that, at runtime, if the regex fails to match any group, I can test for null results to explicitly avoid the operations on the non-existing logfields array entries - not sure how advantageous that would be, but certainly doesn't seem wrong.

However, more advantageous would be to avoid the regex operation itself, which we could avoid if the content of the target field is empty. Please share other ideas you might have to avoid the regex operation at runtime.

Reply:
@jakesmith please see latest commit

Reply:
I think I did misunderstand, believing at the time of comment that generateHPCCLogColumnstAllColumns could be called for other fields, not just LogMessage or LogEntry, the code is clearer now.